Core/MIPS/ARM/ArmJit.cpp

```diff
@@ -602,19 +598,17 @@ void ArmJit::RestoreRoundingMode(bool force) {
 void ArmJit::ApplyRoundingMode(bool force) {
 	// NOTE: Must not destroy R0.
 	// If the game has never set an interesting rounding mode, we can safely skip this.
-	if (g_Config.bSetRoundingMode && (force || !g_Config.bForceFlushToZero || js.hasSetRounding)) {
+	if (force || js.hasSetRounding) {
 		LDR(SCRATCHREG2, CTXREG, offsetof(MIPSState, fcr31));
-		if (!g_Config.bForceFlushToZero) {
-			TST(SCRATCHREG2, AssumeMakeOperand2(1 << 24));
-			AND(SCRATCHREG2, SCRATCHREG2, Operand2(3));
-			SetCC(CC_NEQ);
-			ADD(SCRATCHREG2, SCRATCHREG2, Operand2(4));
-			SetCC(CC_AL);
-			// We can only skip if the rounding mode is zero and flush is set.
-			CMP(SCRATCHREG2, Operand2(4));
-		} else {
-			ANDS(SCRATCHREG2, SCRATCHREG2, Operand2(3));
-		}
+
+		TST(SCRATCHREG2, AssumeMakeOperand2(1 << 24));
+		AND(SCRATCHREG2, SCRATCHREG2, Operand2(3));
+		SetCC(CC_NEQ);
+		ADD(SCRATCHREG2, SCRATCHREG2, Operand2(4));
+		SetCC(CC_AL);
+		// We can only skip if the rounding mode is zero and flush is not set.
+		CMP(SCRATCHREG2, Operand2(3));
```

Comment:
Sorry for missing, this turned into 3 somehow (as you noted in 30555f31ca75e2f9f8e946d2c99eff5cc287d9c1)?

-[Unknown]